### PR TITLE
ARROW-4207: [Gandiva] [GLib] Add support for IfNode

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/gandiva-glib/node.cpp [hunk 1]
```diff
@@ -1180,6 +1182,171 @@ ggandiva_string_literal_node_get_value(GGandivaStringLiteralNode *node)
...
+ *
+ * Returns: A newly created #GGandivaIfNode.
+ *
+ * Since: 0.12.0
+ */
+GGandivaIfNode *
+ggandiva_if_node_new(GGandivaNode *condition_node,
+                     GGandivaNode *then_node,
+                     GGandivaNode *else_node,
+                     GArrowDataType *return_type)
```
Comment (kou) [inline c_glib/gandiva-glib/node.cpp]: Could you add `GError **error` and set an error when `MakeIf` returns `nullptr`?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comment?

File: c_glib/gandiva-glib/node.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/gandiva-glib/node.cpp]: Yes. I'll add it.

**Review by kou — APPROVED** [review]

**Comment by kou** [comment]
Rust build failure is unrelated. I'll merge this.

**Pull request closed without merging**